up42/workflow.py

```diff
@@ -378,9 +382,26 @@ def construct_parameters(
                 input_parameters[data_block_name]["ids"] = scene_ids
                 input_parameters[data_block_name]["limit"] = len(scene_ids)
                 input_parameters[data_block_name].pop("time")
-            elif start_date is not None and end_date is not None:
-                time = f"{start_date}T00:00:00Z/{end_date}T23:59:59Z"
-                input_parameters[data_block_name]["time"] = time
+            elif start_date is not None or end_date is not None:
+                try:
+                    start_date_iso: datetime = datetime.fromisoformat(start_date)  # type: ignore
+                    end_date_iso: datetime = datetime.fromisoformat(end_date)  # type: ignore
```

Comment:
What if it's already datetime object? Will this always fail?

Reply:
Yes. But should maybe change this logic and instead ask for the different types to be more explicit in the code.